libbeat/publisher/queue/diskqueue/serialize_test.go

```diff
@@ -53,17 +89,29 @@ func TestSerialize(t *testing.T) {
 
 		// Use decoder to decode the serialized bytes.
 		decoder := newEventDecoder()
+		decoder.serializationFormat = test.format
 		buf := decoder.Buffer(len(serialized))
 		copy(buf, serialized)
 		decoded, err := decoder.Decode()
 		if err != nil {
 			t.Fatalf("[%v] Couldn't decode serialized data: %v", test.name, err)
 		}
 
-		decodedValue, err := decoded.Content.Fields.GetValue("test_field")
-		if err != nil {
-			t.Fatalf("[%v] Couldn't get field 'test_field': %v", test.name, err)
+		switch test.format {
+		case SerializationCBOR:
+			event, ok := decoded.(publisher.Event)
+			assert.True(t, ok)
+			decodedValue, err := event.Content.Fields.GetValue("test_field")
+			if err != nil {
+				t.Fatalf("[%v] Couldn't get field 'test_field': %v", test.name, err)
+			}
+			assert.Equal(t, test.value, decodedValue)
```

Comment:
[Question]

Why does the decoding error stops the test execution but the wrong value does not?

[Suggestion]
I know you're extending the test to work with the new feature, but  maybe it could be refactored to use table tests, so if one test case fails, that test can stop on failure, but the other test cases still run.

Reply:
updated, see if this looks more like what you were expecting